sci-libs/primegen/primegen-0.97.recipe

```diff
@@ -0,0 +1,81 @@
+SUMMARY="Fast C/C++ prime number generator library"
+DESCRIPTION="primegen is a small, fast library to generate prime numbers \
+in order. It generates the 50847534 primes up to 1000000000 in just 8 \
+seconds on a Pentium II-350; it prints them in decimal in just 35 seconds. \
+primegen can generate primes up to 1000000000000000, although it is not \
+optimized for primes past 32 bits. It uses the Sieve of Atkin instead of the \
+traditional Sieve of Eratosthenes."
+HOMEPAGE="https://cr.yp.to/primegen.html"
+COPYRIGHT="1999, D. J. Bernstein"
+LICENSE="Public Domain"
+REVISION="2"
+SOURCE_URI="https://cr.yp.to/primegen/primegen-$portVersion.tar.gz"
+CHECKSUM_SHA256="54285baf8eed9e421ff2220a2112d38cfb20c1ebef6014ef3f0004c22c95f40d"
+SOURCE_DIR="primegen-$portVersion"
+PATCHES="primegen-$portVersion.patchset"
+
+ARCHITECTURES="?x86_gcc2 x86 x86_64"
+
+PROVIDES="
+	primegen = $portVersion
+	cmd:primegaps = $portVersion
+	cmd:primes = $portVersion
+	cmd:primespeed = $portVersion
+	"
```

Comment:
Since you have made the changes to build the shared lib, we need to declare it in `PROVIDES` with:
```
	lib:libprimegen
```

BTW, with the current recipe the 3 runtimes (primegaps, primes and primespeed) are being linked against the static lib.
It would be nice to make them use the shared lib. I'll let you dig that ;-)

Reply:
The way that the applications are compiled is rather complicated... Right now, I make minimal changes to the Makefile and most of the application is compiled in order to keep things simple... If the programs compiled using the shared library instead, it might mean a lot of changes...

Reply:
Don't worry. In case the changes are not trivial then we can keep the static linking.

Regarging your questions about the tests:
> The >32-bit test takes a while, so I decided to only keep the smallest one.

Perfect.

> When and where is the test run? If it's after the INSTALL function, isn't using primes to test the correct installation of the primes program part of the testing? 😛

The TEST function is only called by haikuporter if you pass `--test` to it.
But you can only call `haikuporter --test primegen` _after_ a successful build.

BTW, it is recommended to use the ` --strict-policy` (or its short `-S` syntax) option during the build.
That's how I found that the `lib:libprimegen` was missing ;-)